independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/BuilderInfo.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.bootstrap.resolver;
+
+import java.util.List;
+
+import io.quarkus.bootstrap.BootstrapDependencyProcessingException;
+import io.quarkus.bootstrap.BootstrapException;
+import io.quarkus.bootstrap.model.AppDependency;
+
+/**
+ * Provides access to information from builders such as Maven and Gradle.
+ */
+public interface BuilderInfo extends AutoCloseable {
```

Comment:
I think we should keep the workspace API and discovery separate from the artifact and AppModel resolution. Perhaps we could split it in smaller tasks. E.g. have a Gradle LocalProject discovery mechanism. And then improve the existing AppModelGradleResolver.

Reply:
I am not sure I understand what you mean.
Would you like BuilderInfos split into LocalProjects and ArtifactResolvers, with a new separate type ArtifactResolver that defines the resolution call?

Reply:
Yes, I wanted to separate the workspace discovery from the rest. There already is AppModelResolver interface and a limited Gradle build-time impl of it. I think we could take that as the base. What do you think?